src/ddspdrum/parameter.py

```diff
@@ -67,10 +75,8 @@ def set_value_0to1(self, new_value):
         ----------
         new_value (float)   :   value to update parameter with, in the range [0,1]
         """
-        new_value = np.clip(new_value, 0, 1)
-
-        if new_value != 0 and self.scale != 1:
-            new_value = np.exp2(np.log2(new_value) / self.scale)
+        if new_value != 0 and self.curve != 1:
+            new_value = np.exp2(np.log2(new_value) / self.curve)
```

Comment:
No asserting here either. I assume we want to avoid throwing an assertion error during learning in the case it is slightly above or below 0 or 1?

Reply:
Yes. The learner module should be responsible for preclipping.

See my suggestion here: https://github.com/turian/ddsp-drum/pull/17#discussion_r554015001
About how we should separate responsibilities